### ISSUE TITLE
Compatibility issue with spree 3.6/3.7

### ISSUE THREAD

**Issue by dbbooke12** [mollie/spree-mollie-gateway#81]
Seems to be a comtability issue with spree 3.6 and 3.7 with spree_mollie_gateway 3.2.

I’ve installed a fresh spree 3.7.3, Rails 5.2.3, Ruby 2.5.3 and I’m getting both these errors when install the gem and generator. I’ve noticed tickets in the past regarding this issue but doesn’t seem to be fixed? Not sure.

![B4E05D65-2C67-4FE6-A9E1-46640623754C](https://user-images.githubusercontent.com/54075071/64988480-61d0d400-d880-11e9-9596-2dd9f95cfc83.png)
<img width="1257" alt="10DBA7C1-2E7E-45F3-9B56-63904C60FEDE" src="https://user-images.githubusercontent.com/54075071/64988482-61d0d400-d880-11e9-9103-7d622b886494.png">

**Comment by vernondegoede** [comment]
@FabianOudhaarlem Can you take a look? This probably happens because there's >= 1 payment(s) in the DB already, but the Spree Mollie Gateway isn't enabled yet (which is also not possible, as the admin breaks).

@dbbooke12 Can you confirm this theory?

**Comment by Oldharlem** [comment]
@vernondegoede I will have a look

**Comment by dbbooke12** [comment]
That's exactly what it was @vernondegoede . When you install spree by default they run sample data during the generators and i think it generates some sample orders. 

When I disabled the sample data/seeds I no longer got the error in admin/orders for "undefined method status". However, during checkout the payment forms still does not show up. I have set up all shipping methods, countries, states for billling as well but still no form.

**Comment by vernondegoede** [comment]
Thanks for the confirmation @dbbooke12! Did you also set up the payment method within the admin? And set the correct API key, specify visibility ("Both"), etc?

**Comment by dbbooke12** [comment]
@vernondegoede That is correct. I have setup the correct API keys and vibisility. I have provided you with an image.

<img width="1259" alt="Screen Shot 2019-09-18 at 11 21 08 AM" src="https://user-images.githubusercontent.com/54075071/65174845-85258b80-da06-11e9-8175-753d4b1398bb.png">

**Comment by vernondegoede** [comment]
@dbbooke12 Thanks a lot! Very useful.

Don’t forget to reset your test API key in our Dashboard, as it’s now publicly available.

**Comment by dbbooke12** [comment]
First thing I did haha. Thanks! Any idea what could be causing this? @vernondegoede

**Comment by dbbooke12** [comment]
any update on this

**Comment by Oldharlem** [comment]
Hi @dbbooke12 
I have looked into the issue and found conflicting code on the admin order index page. The status method of the mollie payment source was being called on all orders independent of payment method. PR #82 fixes the issue. 

The other issue you were experiencing on the checkout does not occur for me. I have created a project with mollie installed. Spree 3.7.3 ruby 2.5.1 and rails 5.2.3
I linked it below.

[spree 3.7.3 mollie](https://github.com/Oldharlem/spree_37_issue_81)
